### PR TITLE
Add Life Progress extension

### PR DESCRIPTION
**Pull request by koinzhang**
View your life progress, get the word of the day

## Description

**Main Feature**

- **Life Progress**: View your life progress, get the word of the day
- **Find Cake**: If it's your birthday, happy birthday to you! Go find the hidden cake.

**Preference:**

- Birthday: Your birthday
  - If you want to **Find Cake** right away, you can set the month and day of your birthday to today
- Week Start: Weekly start date
- Icon Theme: Date icon style theme

## Screencast

 
https://user-images.githubusercontent.com/36128970/161684626-5c7ed0d9-961c-4521-9787-35562e70d8b9.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `metadata` folder

### CODE REVIEW

**Comment by koinzhang** [comment]
Hi @pernielsentikaer👋. Thank you for review.

> It seems like the quote takes a few seconds to load, is there a way to cache this (even just for a day)

Here may be due to geographical problems, the server of this URL is in China, so the access speed may be slower in the United States and other places.
Update: Added a preference for whether or not to show the sentence of the day. If you don't want to show the sentence of the day, you can turn off this option

> Does this item give any value? I'm not even sure what it is, maybe days to the new year? Should maybe just be an integer

Here is where I overlooked the decimal problem, which has now been fixed.

> What does the M mean here?

There does seem to be some confusion here, so I've removed them and just kept the numbers.

> Start words with capital letter, like: spent > Spent

Update: All sentences with all initial letters capitalized.

File: extensions/life-progress/package.json [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  ],
+  "license": "MIT",
+  "preferences": [
+    {
+      "name": "showDailyWord",
+      "type": "checkbox",
+      "required": true,
+      "title": "UI Preference",
+      "label": "Show Daily Word",
+      "description": "Display the sentence of the day. There may be some delay, so if you can't stand it, you can choose to turn this option off.",
```
Comment (pernielsentikaer) [inline extensions/life-progress/package.json:19]:
```suggestion
      "label": "Show Quote of the Day",
      "description": "Show quote of the day",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @koinzhang 👋 

I think we're slowly getting more and more done, awesome 🔥 

I have some new feedback:

- There should be some validation on the birthday, if I input `19285-08-14` it gives some weird numbers, it should be possible to validate age 😄 

CleanShot 2022-04-20 at 20 23 05@2x

- Seems like there is a bug when it's your birthday

https://user-images.githubusercontent.com/38033107/164298006-8224ada5-0734-48f4-b6e7-1de9cd5baf5d.mp4

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by koinzhang** [comment]
Hi @pernielsentikaer👋. Thank for your review. Here are latest changes.

> There should be some validation on the birthday, if I input 19285-08-14 it gives some weird numbers, it should be possible to validate age 

Since I let the user fill in the birthday date in the preferences, I can't regulate what the user fills in. However, I am going to validate the user input and set the birthday date as the default if it is obviously invalid.
Update:When the user enters an illegitimate birthday, it will be reset to the default value of 1995-01-01 and `Failure Toast` will pop up to remind the user when they enter the extension

> Seems like there is a bug when it's your birthday

Update: The action of all list items will become "Rummage Here", and the list will not be refreshed after the cake is found.

**Comment by pernielsentikaer** [comment]
Hi @koinzhang 👋 

I think we're close, last round hopefully 😄 

- The validation seems to work fine but what about changing so the birthdate minus now should be less than 110 instead.
- Maybe just show a [List.EmptyView](https://developers.raycast.com/api-reference/user-interface/list#list.emptyview) instead, something like:

CleanShot 2022-04-21 at 08 48 02@2x

**Comment by koinzhang** [comment]
Hi @pernielsentikaer 👋. Thank for your review and advice.

> The validation seems to work fine but what about changing so the birthdate minus now should be less than 110 instead.

Update: Birthday date verification is now limited to the current date to the last 110 years.

> Maybe just show a [List.EmptyView](https://developers.raycast.com/api-reference/user-interface/list#list.emptyview) instead, something like:

That's a good idea, I've changed it in the latest commit.

**Comment by pernielsentikaer** [comment]
I think we're ready to merge if you don't have anything else? 😄

**Comment by koinzhang** [comment]
Ok, thanks for the review.

**Review by pernielsentikaer — APPROVED** [review]
Let's get it merged 🎉

**Comment by minhashemi** [comment]
Hello there,
Some countries start their week in Saturday. Please consider adding all weekdays to be able to choose from in the next update.
waiting for next amazing update on the Life progress extension

**Comment by koinzhang** [comment]
Hi @minhashemi Thank you for your suggestions.
I will try to update and will keep you informed of any new developments.

**Comment by koinzhang** [comment]
Hi @minhashemi.
I have submitted an updated [PR](https://github.com/raycast/extensions/pull/1835) for Life Progress. please look forward to reviewing and approving it.
Updates:
- Add feature **Countdown Date**: Now you can add custom countdown dates
- Add progress bar display and empty view
- Add _Open Preference_ action
- Add _Birthday Every Day_, _Progress Bar Symbol_ preferences
- Update _Week Start_ preference: now you can choose any day as start of week
image
Metadata-3

**Pull request merged**